Take input striding for conv fusion op  based on eager output

XiaobingSuper opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #88988
* __->__ #88864


As https://github.com/pytorch/pytorch/pull/88706, we also change the input stride check using eager output.

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

XiaobingSuper committed:
Take input striding for conv fusion op  based on eager output

[ghstack-poisoned]

XiaobingSuper committed:
Update on "Take input striding for conv fusion op  based on eager output"



As https://github.com/pytorch/pytorch/pull/88706, we also change the input stride check using eager output.

cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

XiaobingSuper committed:
Update on "Take input striding for conv fusion op  based on eager output"



As https://github.com/pytorch/pytorch/pull/88706, we also change the input stride check using eager output.

cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

XiaobingSuper committed:
Update on "Take input striding for conv fusion op  based on eager output"



As https://github.com/pytorch/pytorch/pull/88706, we also change the input stride check using eager output.

cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

XiaobingSuper committed:
Update on "Take input striding for conv fusion op  based on eager output"



As https://github.com/pytorch/pytorch/pull/88706, we also change the input stride check using eager output.

cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

XiaobingSuper committed:
Update on "Take input striding for conv fusion op  based on eager output"



As https://github.com/pytorch/pytorch/pull/88706, we also change the input stride check using eager output.

cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

XiaobingSuper committed:
Update on "Take input striding for conv fusion op  based on eager output"



As https://github.com/pytorch/pytorch/pull/88706, we also change the input stride check using eager output.

cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

jgong5 commented on torch/_inductor/ir.py:
```diff
@@ -3346,59 +3346,47 @@ def _prepare_convolution_fusion_create(
-    else:
-        stride_order = list(reversed(range(len(output_size))))
+    x = cls.require_stride_order(x, req_stride_order)
+    weight = cls.require_stride_order(weight, req_stride_order)
```
Should keep weight layout as in https://github.com/pytorch/pytorch/pull/88891?

XiaobingSuper replied:
Yes, changed.

XiaobingSuper committed:
Update on "Take input striding for conv fusion op  based on eager output"



As https://github.com/pytorch/pytorch/pull/88706, we also change the input stride check using eager output.

cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

XiaobingSuper commented:
@pytorchbot merge

XiaobingSuper commented:
@pytorchbot merge
